src/main.ts

```diff
@@ -222,24 +197,69 @@ app.on('activate', () => {
   }
 })
 
-if (!app.isDefaultProtocolClient(pkg.protocolSchemes.native)) {
-  // Define custom protocol handler. Deep linking works on packaged versions of the application!
-  app.setAsDefaultProtocolClient(pkg.protocolSchemes.native)
-}
+registerCustomProtocol()
 
 app.on('will-finish-launching', function () {
   // Protocol handler for osx
   app.on('open-url', function (event, url) {
     event.preventDefault()
-    deeplinkingUrl = url
-    if (mainWindow) {
-      redirectDeepLinkingUrl(deeplinkingUrl.toString(), mainWindow)
-    }
+    deepLinkingUrl = url
+    redirectDeepLinkingUrl(deepLinkingUrl, mainWindow)
   })
 })
 
 ipcMain.on('app-ready', () => {
-  if (mainWindow && deeplinkingUrl) {
-    redirectDeepLinkingUrl(deeplinkingUrl.toString(), mainWindow)
+  if (app.isPackaged) {
+    const feedURL = `${pkg.hostUrl}/${pkg.repoName}/${process.platform}-${
+      process.arch
+    }/${app.getVersion()}`
+
+    autoUpdater.setFeedURL({
+      url: feedURL,
+      serverType: 'default',
+    })
+    autoUpdater.checkForUpdates()
+
+    const fourHours = 4 * 60 * 60 * 1000
+    setInterval(() => {
+      autoUpdater.checkForUpdates()
+    }, fourHours)
   }
+
+  redirectDeepLinkingUrl(deepLinkingUrl, mainWindow)
+})
+
+ipcMain.on('restart_app', () => {
+  autoUpdater.quitAndInstall()
+})
+
+autoUpdater.on('checking-for-update', () => {
+  console.log('checking-for-update')
+})
+
+autoUpdater.on('update-available', () => {
+  console.log('update-available')
+})
+
+autoUpdater.on(
+  'update-downloaded',
+  (event, releaseNotes, releaseName, updateURL) => {
+    if (mainWindow && mainWindow.webContents) {
+      mainWindow.webContents.send('update_downloaded')
+    }
+    console.log('update-downloaded', {
+      event,
+      releaseNotes,
+      releaseName,
+      updateURL,
+    })
+  },
+)
+
+autoUpdater.on('update-not-available', () => {
+  console.log('update-not-available')
+})
```

Comment:
Same as another comment about `console.log`. Please change it everywhere

Reply:
Updated